core/src/main/scala/org/apache/spark/sql/rapids/tool/util/EventUtils.scala

```diff
@@ -93,6 +94,17 @@ object EventUtils extends Logging {
     properties.getOrElse(propKey, defValue).equals(targetValue)
   }
 
+  def readRootIDFromSQLStartEvent(event: SparkListenerSQLExecutionStart) : Option[Long] = {
+    Try(Option(rootExecutionIdField.get(event)).map(_.asInstanceOf[Option[Long]]).
```

Comment:
This looked slightly confusing to me. Since `rootExecutionIdField` is already an Option, we should not wrap it again followed by flatten.

 Can we refactor as:

```suggestion
    Try(rootExecutionIdField.get(event).asInstanceOf[Option[Long]]).getOrElse(None)
```

Reply:
Can we add a comment why are we using the dynamic reflection API?